### PR TITLE
Universal Router

### PR DESCRIPTION
**Pull request by cartogram**
## Description

Fixes https://github.com/Shopify/quilt/issues/892

This PR adds the initial `universal-router` package. This package just exports mostly-as-is components from `react-router`, but provides a common location for apps to import router components as we swap out the implementation for something inline with #724.

I'm going to release this pre-`1.0.0` and so we can iterate on docs quickly as we go.

### CODE REVIEW

File: packages/universal-router/src/Redirect.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import {withRouter, RouteComponentProps} from 'react-router';
+import {Redirect as NetworkRedirect} from '@shopify/react-network';
+
+interface Props {
+  url: string;
+}
+
+type ComposedProps = Props & RouteComponentProps<{}, {}>;
+
+class Redirect extends React.Component<ComposedProps> {
```
Comment (TheMallen) [inline packages/universal-router/src/Redirect.tsx]: Y U no hook

File: packages/universal-router/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+[![Build Status](https://travis-ci.org/Shopify/quilt.svg?branch=master)](https://travis-ci.org/Shopify/quilt)
+[![License: MIT](https://img.shields.io/badge/License-MIT-green.svg)](LICENSE.md) [![npm version](https://badge.fury.io/js/%40shopify%2Funiversal-router.svg)](https://badge.fury.io/js/%40shopify%2Funiversal-router.svg) [![npm bundle size (minified + gzip)](https://img.shields.io/bundlephobia/minzip/@shopify/universal-router.svg)](https://img.shields.io/bundlephobia/minzip/@shopify/universal-router.svg)
+
+A universal router for React.
+
+## Installation
+
+```bash
+$ yarn add @shopify/universal-router
```
Comment (lemonmade) [inline packages/universal-router/README.md]: `react-router`?

File: packages/universal-router/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "@shopify/universal-router",
+  "version": "0.0.0",
```
Comment (lemonmade) [inline packages/universal-router/package.json]: Let's keep this sub-1.0 until we have our own thing, and make sure the docs are clear about that

File: packages/universal-router/src/utilities.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export function isServer() {
```
Comment (lemonmade) [inline packages/universal-router/src/utilities.ts]: I assume this is just for mocking in tests?

File: packages/universal-router/package.json (hunk 3 above)
Comment (cartogram) [reply packages/universal-router/package.json]: yeah I'll add a note/banner to the readme.

File: packages/universal-router/src/utilities.ts (hunk 4 above)
Comment (cartogram) [reply packages/universal-router/src/utilities.ts]: Yeah, thats right.

**Pull request merged**